src/JSConfetti.ts

```diff
@@ -4,17 +4,59 @@ import { createCanvas } from './createCanvas'
 import { normalizeConfettiConfig } from './normalizeConfettiConfig'
 import { IPosition, IJSConfettiConfig, IAddConfettiConfig } from './types'
 
+class ConfettiBatch {
+  private promiseCompletion?: () => void
+  private promise?: Promise<void>
+  private members: ConfettiShape[]
+
+  constructor() {
+    this.members = []
+  }
+
+  getPromise(): Promise<void> {
+    if (!this.promise) {
+      this.promise = new Promise((completionCallback) => this.promiseCompletion = completionCallback)
```

Comment:
maybe we can move this initialization to the `constructor`?

so we don't need conditional statement here and later we don't need optional chaining `this.promiseCompletion?.()`

Reply:
Sure. It didn't really solve the optional chaining however. See my commit comment:
Original desire was to 'only create the promise if someone actually uses it'. Except that is not really how its used -- poor choice on my part with the orignal code.

However, moving it to always create doesn't remove the ugly wart of optional chaining. This is 'cause the flow from the promise constructor is passed a function, and flow analsysis doesn't *know* the field is assigned to, so demands optionality.

One option would be to assign a function to it by default, that is then overwritten, and all would be good. However that feels noisy given the allocation of the function that is never used.